db_manager/dumper.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+import argparse
+import os
+import subprocess
+from datetime import date
+import psycopg2
+import psycopg2.extras
+
+class Dumper():
+    """This class is used to dump and restore a Postgres database."""
+
+    def __init__(self, pg_service, file):
+        self.connection = psycopg2.connect("service={}".format(pg_service))
+        self.file = file
+
+    def __get_dbname(self):
+        return self.connection.get_dsn_parameters()['dbname']
+
+    def __get_dbuser(self):
+        return self.connection.get_dsn_parameters()['user']
+
+    def __print_connection_info(self):
+
+        cursor = self.connection.cursor()
+        cursor.execute('select version() as version '
+                       ', current_database() as db '
+                       ', current_user as user '
+                       ', now()::text as query_time '
+                       )
+        result = cursor.fetchone()
+        print(result)
+        print(self.connection.get_dsn_parameters())
+        cursor.close()
+
+    def pg_backup(self):
+        #TODO test on Windows and OSX
+
+        #pg_dump_exe = 'C:\\Program Files\\PostgreSQL\\9.3\\bin\\pg_dump.exe'
+        pg_dump_exe = 'pg_dump'
+
+        command = []
+        command.append(pg_dump_exe)
+        command.append('-Fc')
+        command.append('-U')
+        command.append(self.__get_dbuser())
+        command.append('-f')
+        command.append(self.file)
+        command.append(self.__get_dbname())
+
+        try:
+            subprocess.check_output(command, stderr=subprocess.STDOUT)
+
+        except subprocess.CalledProcessError as e:
+            print('pg_dump failed', e.output)
+            raise SystemExit(1)
```

Comment:
That sounds a bit hard, I can imagine we will want to run this from inside QGIS in some user interface the future.